primo/data_parser/well_data.py

```diff
@@ -0,0 +1,387 @@
+        "data",  # DataFrame containing well data
+        # Private variables
+        "_col_names",  # Pointer to WellDataColumnNames object
+        "_removed_rows",  # dict containing list of rows removed
+        "_num_wells_stats",  # dict containing statistics on num. wells
+    )
+
+    def __init__(
+        self,
+        filename: str,
+        column_names: WellDataColumnNames,
+        **kwargs,
+    ) -> None:
+        """
+        Reads well data and performs initial high-level data processing.
+
+        Parameters
+        ----------
+        filename : str
+            Name of the file containing the well data. Currently, only
+            .xlsx, .xls, and .csv formats are supported.
+
+        column_names : WellDataColumnNames
+            A WellDataColumnNames object containing the names of various columns
+        """
+        # Import columns whose column names have been provided.
+        LOGGER.info("Reading the well data from the input file.")
+        if filename.split(".")[-1] in ["xlsx", "xls"]:
+            self.data = pd.read_excel(
+                filename,
+                sheet_name=kwargs.pop("sheet", 0),
+                usecols=column_names.values(),
+                # Store well ids as strings
+                dtype={column_names.well_id: str},
+            )
+
+        elif filename.split(".")[-1] == "csv":
+            self.data = pd.read_csv(
+                filename,
+                usecols=column_names.values(),
+                # Store well ids as strings
+                dtype={column_names.well_id: str},
+            )
+
+        else:
+            raise_exception(
+                "Unsupported input file format. Only .xlsx, .xls, and .csv are supported.",
+                TypeError,
+            )
+
+        # Updating the `index` to keep it consistent with the row numbers in
+        # xlsx, xls, or csv files
+        self.data.index += 2
+
+        # Read input options
+        self.config = WELL_DATA_CONFIG(kwargs)
+        self._removed_rows = {}
+        self._col_names = column_names
+        # Store number of wells in the input data
+        num_wells_input = self.data.shape[0]
+        self._num_wells_stats = {"input_data": num_wells_input}
+
+        LOGGER.info("Finished reading the well data. Beginning to process it.")
+
+        # Start processing input data
+        # self._process_input_data()  # Uncomment in the next PR
+
+        num_wells_processed = self.data.shape[0]
+        self._num_wells_stats["initial_processing"] = num_wells_processed
+        LOGGER.info("Finished preliminary processing.")
+
+    def __contains__(self, val):
+        """Checks if a column is available in the well data"""
+        return val in self.data.columns
+
+    def __iter__(self):
+        """Iterate over all rows of the well data"""
+        return iter(self.data.index)
+
+    @property
+    def get_removed_wells(self):
+        """Yields the list of wells removed from the data set"""
```

Comment:
Minor but would recommend changing docstring to "Returns" and not "Yields" since this is not a generator function